doc/manual/src/glossary.md

```diff
@@ -208,44 +208,85 @@
 
     See [installables](./command-ref/new-cli/nix.md#installables) for [`nix` commands](./command-ref/new-cli/nix.md) (experimental) for details.
 
-  - [build system]{#gloss-build-system}
+  - [platform]{#gloss-platform}
 
-    [build system]: #gloss-build-system
+    [platform]: #gloss-platform
 
-    A representation of the processor architecture and platform that performs the build or compilation, typically a dash-separated string.
+    A set of execution environments capable of executing some set of software. A platform is typically described by attributes such as instruction set architecture, operating system, and application binary interface.
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    Nix is not intimately aware of all possible platforms.
+    Rather, it abstracts the platform into
+     - [system type]{#gloss-system-type} string
+     - [system features](command-ref/conf-file.md#system-features), to a lesser extent, as these are often thought of as additions to a platform, rather being part of the platform definition.
 
-    A program that is produced by a [derivation] is built with a [build system] that can be executed by the [derivation system](language/derivations.md#attr-system).
-    After it has been built, the [build system] is no longer relevant.
+  - [system type]:{#gloss-system-type}
 
-  - [host system]{#gloss-host-system}
+    An identifier for a specific [platform] or (in rare cases) a class of compatible platforms, such as `x86_64-v3-linux`.
 
-    [host system]: #gloss-host-system
+    To build a derivation locally, the [derivation `system` field](language/derivations.md#attr-system) must occur in the local Nix installation's [`system`](command-ref/conf-file.md#conf-system) or [`extra-platforms`](command-ref/conf-file.md#conf-extra-platforms) settings, among other rules.
+    Otherwise, a [remote build](advanced-topics/distributed-builds.md) is needed.
 
-    A representation of the processor architecture and platform where build products will run on, typically a dash-separated string.
+  - [build platform]{#gloss-build-platform}
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    [build platform]: #gloss-build-platform
 
-    Programs that run natively in a [derivation] must be configured with a [host system] that can by executed by the [derivation system](language/derivations.md#attr-system).
+    The [platform] that performs the build or compilation, typically a dash-separated string.
 
-  - [target system]{#gloss-target-system}
+    A program that is produced by a [derivation] has a [build configuration] with a [build platform] that can be executed by the [derivation system](language/derivations.md#attr-system).
+    After it has been built, the [build platform] is no longer relevant.
```

Comment:
No need to link the term to itself. All the links are already a strain to read.

As discussed today:

```suggestion
    A program that is produced by a [derivation] must have a [build configuration] where the build platform matches the [`system` field of that derivation](language/derivations.md#attr-system).
    After it has been built, the build platform is no longer relevant.
```

Same for the other occurrences.